bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -223,6 +225,61 @@ def jit_function(self, func, cols_list, extra_args):
         FunctionJitter.function_cache[self.func.__name__] = (self.func_call, self.func_sign)
         return self.func_call
 
+def _convert_to_vector(args):
+    """
+    Converts a Python list of strings into an std::vector before passing such
+    argument to a Filter operation.
+    The purpose of this function is to workaround issue #10092 until there is
+    a proper fix in cppyy.
+    Arguments:
+        args: arguments passed to Filter, possibly including the list of column
+    names
+    Returns:
+        Tuple with arguments, possibly replacing a list of column names by a
+    vector
+    """
+
+    import ROOT
+
+    if not isinstance(args[0], list): return args
+
+    try:
+        v = ROOT.std.vector['std::string'](args[0])
+    except TypeError:
+        raise TypeError(
+            f"The list of columns of a Filter operation can only contain strings. Please check: {args[0]}")
+
+    return v, *args[1:]
+
+def _handle_cpp_callables(func, original_template, *args):
+    """
+    Checks whether the callable `func` is a cppyy proxy of one of these:
+    1. C++ functor
+    2. std::function
+
+    The cases above are supported by cppyy, so we can just invoke the original
+    cppyy TemplateProxy (Filter or Define) with the callable as argument.
+
+    Prior to the invocation of the original cppyy TemplateProxy, though, we
+    need to explicitly instantiate the template using the type of the `func`
+    callable. Implicit instantiation won't work since the original
+    implementation of Filter and Define is replaced by a pythonization.
+
+    Arguments:
+        func: callable to be checked
+        original_rdf_method: cppyy proxy for Filter or Define
+        args: arguments passed by the user to Filter or Define
+    Returns:
+        RDataFrame: RDataFrame node if `func` can be classified in one of the
+    three cases above, None otherwise
+    """
+
+    is_cpp_functor  = lambda : type(getattr(func, '__call__', None)) == libcppyy.CPPOverload
+    is_std_function = lambda : isinstance(getattr(func, 'target_type', None), libcppyy.CPPOverload)
+
+    if is_cpp_functor() or is_std_function():
+        return original_template[str(type(func)), 0](*args)
```

Comment:
What is the need for the second argument to the template `0`?

Reply:
I'm also wondering how `str(type(func))` works. Take this example:
```python
>>> import ROOT
>>> ROOT.gInterpreter.Declare("""
... std::function<ULong64_t(ULong64_t)> myfun = [](ULong64_t l) { return l*l; };
... """)
True
>>> 
>>> ROOT.myfun
<cppyy.gbl.std.function<ULong64_t(ULong64_t)> object at 0x7fdd6ce51020>
>>> type(ROOT.myfun)
<class cppyy.gbl.std.function<ULong64_t(ULong64_t)> at 0x55cdf13377b0>
>>> str(type(ROOT.myfun))
'<class cppyy.gbl.std.function<ULong64_t(ULong64_t)> at 0x55cdf13377b0>
```

This means we are instantiating the template with that string which includes a lot of extra info, including the memory address where the Python class proxy is defined. The fact this works looks super weird :/

Reply:
> What is the need for the second argument to the template 0?

It is indeed redundant, thanks for the catch!

> This means we are instantiating the template with that string which includes a lot of extra info, including the memory address where the Python class proxy is defined. The fact this works looks super weird :/

Yes it does work, my guess is that cppyy is able to parse it's own type string formatting. But I agree that removing the `str` part will make it simpler and still works, so I will go for that, thanks.